tfhe/src/core_crypto/commons/numeric/unsigned.rs

```diff
@@ -228,4 +263,77 @@ mod test {
                 .to_string()
         );
     }
+
+    #[test]
+    fn test_wrapping_add_custom_mod() {
+        let a = u64::MAX;
+        let b = u64::MAX;
+        let custom_modulus_u128 = (1u128 << 64) - (1 << 32) + 1;
+        let custom_modulus = custom_modulus_u128 as u64;
+
+        let a_u128: u128 = a.into();
+        let b_u128: u128 = b.into();
+
+        let expected_res = ((a_u128 + b_u128) % custom_modulus_u128) as u64;
+
+        let res = a.wrapping_add_custom_mod(b, custom_modulus);
+        assert_eq!(expected_res, res);
+
+        const NB_REPS: usize = 100_000_000;
+
+        use rand::Rng;
+        let mut thread_rng = rand::thread_rng();
+        for _ in 0..NB_REPS {
+            let a: u64 = thread_rng.gen();
+            let b: u64 = thread_rng.gen();
```

Comment:
If I understand correctly the test is using inputs which are not within the custom mod ?

Reply:
that's a good point 🤔 I'll update